employee-rostering-backend/src/test/java/org/optaweb/employeerostering/controller/SkillControllerTest.java

```diff
@@ -0,0 +1,254 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.controller;
+
+import java.util.List;
+
+import javax.persistence.EntityNotFoundException;
+import javax.transaction.Transactional;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.optaweb.employeerostering.domain.Skill;
+import org.optaweb.employeerostering.service.SkillService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@Transactional
+public class SkillControllerTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(SkillControllerTest.class);
+
+    @Autowired
+    private SkillService skillService;
+
+    @Test
+    public void getSkillListTest() {
+        Integer tenantId = 1;
+        Integer tenantId2 = 2;
+        String name = "name";
+        String name2 = "name2";
+
+        Skill skill = new Skill(tenantId, name);
+        Skill skill2 = new Skill(tenantId, name2);
+        Skill skill3 = new Skill(tenantId2, name);
+
+        skillService.createSkill(tenantId, skill);
+        skillService.createSkill(tenantId, skill2);
+        skillService.createSkill(tenantId2, skill3);
+
+        List<Skill> skillList = skillService.getSkillList(tenantId);
+        List<Skill> skillList2 = skillService.getSkillList(tenantId2);
+
+        assertEquals(skillList.get(0), skill);
+        assertEquals(skillList.get(1), skill2);
+        assertEquals(skillList2.get(0), skill3);
+    }
+
+    @Test
+    public void getSkillTest() {
+        Integer tenantId = 1;
+        String name = "name";
+
+        Skill skill = new Skill(tenantId, name);
+
+        skillService.createSkill(tenantId, skill);
+
+        Skill returnSkill = skillService.getSkill(tenantId, skill.getId());
+
+        assertEquals(returnSkill.getTenantId(), (Integer) 1);
+        assertEquals(returnSkill.getName(), "name");
+    }
+
+    @Test
+    public void getNonExistentSkillTest() {
+        Integer tenantId = 1;
+        String name = "name";
+
+        Skill skill = new Skill(tenantId, name);
+
+        skillService.createSkill(tenantId, skill);
+
+        try {
+            skillService.getSkill(1, -1L);
+        } catch (EntityNotFoundException e) {
+            String expectedMessage = "No Skill entity found with ID (-1).";
+            assertEquals(expectedMessage, e.getMessage());
```

Comment:
Non Blocker; you can use AssertJ for testing exceptions (see https://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#exception-assertion). In this case, it be

```
assertThatExceptionOfType(EntityNotFoundException.class).isThrownBy(() => skillService.getSkill(1, -1L))
    .withMessage("No Skill entity found with ID (-1).");
```